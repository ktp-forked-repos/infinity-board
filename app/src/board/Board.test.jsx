/* eslint-disable */
/**
 * @jest-environment node
*/

import React from 'react';
import renderer from 'react-test-renderer';
import Board from './board';
import ServerComm from '.././server-comm';
import DraggableBox from './draggable-box';
import ImageBox from './image-box';


// Implementation of websocket mock
const mockUpdateReceived = jest.fn();
const mockSendUpdateMsg = jest.fn();
const mockConnect = jest.fn();

jest.mock('.././server-comm', () => {
  return jest.fn().mockImplementation(() => {
    return ({
      setReceivedUpdateMessageHandler: mockUpdateReceived,
      sendUpdateMessage: mockSendUpdateMsg,
      connect: mockConnect,
    });
  });
});

beforeEach(() => {
  // Clear all instances and calls to constructor and all methods:
  ServerComm.mockClear();
  mockUpdateReceived.mockClear();
  mockSendUpdateMsg.mockClear();
  mockConnect.mockClear();
});


const mockData = {
  _id: 'test',
  elements: {
    testBox: {
      type: 'image',
      state: {
        x: 50,
        y: 50,
        w: 200,
        h: 200,
        z: 1,
        color: '#ffffff',
        text: '',
        src: 'test',
      },
    },
  },
};

const mockSocketMsg = {
  zIndex: 10,
  uuid: 'testBox',
  type: 'image',
  state: {
    x: 100,
    y: 100,
    w: 2000,
    h: 2000,
    z: 10,
    color: '#ffffff',
  }
}


it('renders correctly', () => {
  const tree = renderer.create(<Board data={mockData} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('generates boxes', () => {
  let doc = renderer.create(<Board data={mockData} />);
  let instance = doc.root.instance;
  instance.generateBox('text');
  expect(doc.toJSON()).toMatchSnapshot();
});

it('resizes images', () => {
  let doc = renderer.create(<Board data={mockData} />);
  let instance  = doc.root.instance;
  instance.updateImage('testBox', 750, 450);
  expect(instance.state.boxes.testBox.state.w).toEqual(500);
});

it('receives websocket updates', () => {
  let doc = renderer.create(<Board data={mockData} />);
  let instance  = doc.root.instance;
  instance.onUpdate(mockSocketMsg);
  expect(instance.state.boxes.testBox.state.h).toEqual(2000);
});

it('sends websocket updates', () => {
  let doc = renderer.create(<Board data={mockData} />);
  let instance  = doc.root.instance;
  instance.updateBoardState('testBox', {});
  expect(mockSendUpdateMsg).toHaveBeenCalledTimes(1);
});

it('resizes boxes correctly', () => {
  let doc = renderer.create(<DraggableBox callback={() => {}} uid={'1234'} />);
  let instance  = doc.root.instance;
  expect(instance.getResize(500, 600, 50)).toEqual(50);
});

it('resizes images on render', () => {
  function callbackFunc(uuid, w, h) {
    expect(w).toEqual(0);
    done();
  }
  const doc = renderer.create(<ImageBox w={200} h={200} src={'test'} uid={'1234'} imgCallback={callbackFunc} callback={() => {}} />);
});
