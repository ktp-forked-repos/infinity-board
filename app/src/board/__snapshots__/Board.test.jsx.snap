// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates boxes 1`] = `
<div
  className="View"
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "cursor": undefined,
    }
  }
>
  <div>
    <div
      className="Box"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "cursor": "default",
          "height": "200px",
          "left": "50px",
          "padding": "0px",
          "top": "50px",
          "visibility": "hidden",
          "width": "200px",
          "zIndex": 1,
        }
      }
    >
      <div
        className="Box-image"
        style={
          Object {
            "backgroundImage": "url(test)",
            "opacity": 0,
            "visibility": "visible",
          }
        }
      >
        <img
          alt=""
          onLoad={[Function]}
          src="test"
          style={
            Object {
              "visibility": "hidden",
            }
          }
        />
      </div>
      <div
        className="pull-tab"
        style={
          Object {
            "opacity": 0,
            "zIndex": 3,
          }
        }
      />
    </div>
  </div>
  <div>
    <div
      className="Box"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#ff0000",
          "cursor": "default",
          "height": "160px",
          "left": "50px",
          "padding": "20px",
          "top": "50px",
          "visibility": "visible",
          "width": "160px",
          "zIndex": 1,
        }
      }
    >
      <div
        className="Text-box unselectable"
      >
        
      </div>
      <div
        className="pull-tab"
        style={
          Object {
            "opacity": 0,
            "zIndex": 3,
          }
        }
      />
    </div>
  </div>
  <div
    className="View"
    id="bg"
    style={
      Object {
        "backgroundPosition": "0px 0px",
      }
    }
  />
  <div
    className="Wrapper File-drop"
    style={
      Object {
        "visibility": "hidden",
        "zIndex": -1,
      }
    }
  >
    <div
      className="Drop-text"
    >
      DROP FILE HERE
    </div>
  </div>
  <div
    className="Button-wrapper"
    style={
      Object {
        "zIndex": 2,
      }
    }
  >
    <div
      className="Box-button"
    >
      <button
        className="Box-button home"
        onClick={[Function]}
      >
        HOME
      </button>
    </div>
    <div
      className="Box-button"
      style={
        Object {
          "zIndex": 2,
        }
      }
    >
      <button
        className="Box-button text"
        data-type="text"
        onClick={[Function]}
      >
        TEXT
      </button>
    </div>
    <div
      className="Box-button"
      style={
        Object {
          "zIndex": 2,
        }
      }
    >
      <button
        className="Box-button image"
        data-type="image"
        onClick={[Function]}
      >
        IMAGE
      </button>
      <input
        accept="image/*"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="View"
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "cursor": undefined,
    }
  }
>
  <div>
    <div
      className="Box"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "cursor": "default",
          "height": "200px",
          "left": "50px",
          "padding": "0px",
          "top": "50px",
          "visibility": "hidden",
          "width": "200px",
          "zIndex": 1,
        }
      }
    >
      <div
        className="Box-image"
        style={
          Object {
            "backgroundImage": "url(test)",
            "opacity": 0,
            "visibility": "visible",
          }
        }
      >
        <img
          alt=""
          onLoad={[Function]}
          src="test"
          style={
            Object {
              "visibility": "hidden",
            }
          }
        />
      </div>
      <div
        className="pull-tab"
        style={
          Object {
            "opacity": 0,
            "zIndex": 3,
          }
        }
      />
    </div>
  </div>
  <div
    className="View"
    id="bg"
    style={
      Object {
        "backgroundPosition": "0px 0px",
      }
    }
  />
  <div
    className="Wrapper File-drop"
    style={
      Object {
        "visibility": "hidden",
        "zIndex": -1,
      }
    }
  >
    <div
      className="Drop-text"
    >
      DROP FILE HERE
    </div>
  </div>
  <div
    className="Button-wrapper"
    style={
      Object {
        "zIndex": 2,
      }
    }
  >
    <div
      className="Box-button"
    >
      <button
        className="Box-button home"
        onClick={[Function]}
      >
        HOME
      </button>
    </div>
    <div
      className="Box-button"
      style={
        Object {
          "zIndex": 2,
        }
      }
    >
      <button
        className="Box-button text"
        data-type="text"
        onClick={[Function]}
      >
        TEXT
      </button>
    </div>
    <div
      className="Box-button"
      style={
        Object {
          "zIndex": 2,
        }
      }
    >
      <button
        className="Box-button image"
        data-type="image"
        onClick={[Function]}
      >
        IMAGE
      </button>
      <input
        accept="image/*"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
    </div>
  </div>
</div>
`;
